lib/circuit.js

```diff
@@ -235,89 +238,92 @@ class CircuitBreaker extends EventEmitter {
    * @fires CircuitBreaker#timeout
    */
   fire () {
-    const args = Array.prototype.slice.call(arguments);
-
-    /**
-     * Emitted when the circuit breaker action is executed
-     * @event CircuitBreaker#fire
-     */
-    this.emit('fire', args);
-
-    if (CACHE.get(this) !== undefined) {
-      /**
-       * Emitted when the circuit breaker is using the cache
-       * and finds a value.
-       * @event CircuitBreaker#cacheHit
-       */
-      this.emit('cacheHit');
-      return CACHE.get(this);
-    } else if (this.options.cache) {
-      /**
-       * Emitted when the circuit breaker does not find a value in
-       * the cache, but the cache option is enabled.
-       * @event CircuitBreaker#cacheMiss
-       */
-      this.emit('cacheMiss');
-    }
+    const semaphore = new Semaphore(this.options.capacity);
```

Comment:
This should be a class property. Creating a new semaphore each time the circuit fires doesn't achieve the goal of rate limiting, because the semaphore is only ever used once, and cannot block other code paths from acquiring it.